routes/users.go

```diff
@@ -34,6 +35,17 @@ type registerRequest struct {
 	DateOfBirth *time.Time `json:"dateOfBirth"`
 }
 
+type updateUserRequest struct {
+	NameFirst   *string    `json:"firstName"`
+	NameLast    *string    `json:"lastName"`
+	Email       *string    `json:"email"`
+	PhoneNumber *string    `json:"phoneNumber"`
+	City        *string    `json:"city"`
+	State       *string    `json:"state"`
+	DateOfBirth *time.Time `json:"dateOfBirth"`
+	Favorites   []string   `json:"favorites"`
```

Comment:
maybe have addFavorite and removeFavorite so they don't need to get their original favorites? not 100% necessary but would make front end easier

Reply:
and you can use $addToSet: { $each: [] } and $pullAll: []